File: cpp/cqueue.cpp
#include <iostream>
#include <stack>

using namespace std;

class CQueue {
public:
    stack<int> stack1;
    stack<int> stack2;
    int size;
    CQueue() {

    }

    void AppendTail(int value) {
        while (!stack1.empty()) {
            stack2.push(stack1.top());
            stack1.pop();
        }
        stack1.push(value);
        while (!stack2.empty()) {
            stack1.push(stack2.top());
            stack2.pop();
        }
        size += 1;
    }

    int DeleteHead() {
        if (stack1.empty())
            return -1;
        int head = stack1.top();
        size -= 1;
        stack1.pop();
        return head;
    }
};

int main() {
    CQueue* obj = new CQueue();
    obj->AppendTail(1);
    int param_2 = obj->DeleteHead();
    return 0;
}

File: java/CQueue.java
package com.huaze.shen;

import java.util.Stack;

/**
 * @author Huaze Shen
 * @date 2020-03-09
 */
public class CQueue {
    Stack<Integer> stack1;
    Stack<Integer> stack2;
    int size;

    public CQueue() {
        stack1 = new Stack<Integer>();
        stack2 = new Stack<Integer>();
        size = 0;
    }

    public void appendTail(int value) {
        while (!stack1.isEmpty()) {
            stack2.push(stack1.pop());
        }
        stack1.push(value);
        while (!stack2.isEmpty()) {
            stack1.push(stack2.pop());
        }
        size++;
    }

    public int deleteHead() {
        if (size == 0) {
            return -1;
        }
        size--;
        return stack1.pop();
    }

    public static void main(String[] args) {
        CQueue obj = new CQueue();
        obj.appendTail(1);
        int param_2 = obj.deleteHead();
    }
}